go/tools/builders/compilepkg.go

```diff
@@ -343,6 +351,15 @@ func compileArchive(
 		if err != nil {
 			return err
 		}
+		if coverMode != "" {
+			// Compile original source files, not coverage instrumented, for nogo
+			_, goSrcsNogo, _, err = cgo2(goenv, goSrcsNogo, cgoSrcsNogo, cSrcs, cxxSrcs, objcSrcs, objcxxSrcs, sSrcs, hSrcs, packagePath, packageName, cc, cppFlags, cFlags, cxxFlags, objcFlags, objcxxFlags, ldFlags, cgoExportHPath)
```

Comment:
`goSrcsNogo` and `cgoSrcsNogo` need to be cgo compiled, otherwise cgo enabled targets cannot run static analysis succesfully. Because the source files with `import "C"` are treated as cgoSrcs, and nogo will report type-checking errors on those source files if not cgo compiled.